source/allocator.c

```diff
@@ -34,6 +34,11 @@ bool aws_allocator_is_valid(const struct aws_allocator *alloc) {
     return alloc && AWS_OBJECT_PTR_IS_READABLE(alloc) && alloc->mem_acquire && alloc->mem_release;
 }
 
+struct aws_aligned_allocator_impl {
```

Comment:
use "explicit" or "custom" in this struct name (match the public function name)

Otherwise, one would assume this is for `aws_aligned_allocator()`

Which is extra confusing because both share the s_aligned_malloc() call, and it's tough to explain "you can tell that it's NOT the `aws_aligned_allocator()` because it uses the `aws_aligned_allocator_impl`" 🤯🤯🤯

Reply:
Go with @DmitriyMusatkin 's suggestion to remove the whole impl, and use the point address to store the size_t for the alignment.